lib/metrics/blacklist.go

```diff
@@ -0,0 +1,246 @@
+		return nil, errors.New("cannot create a Blacklist metric without a child")
+	}
+	if c, ok := constructors[config.Blacklist.Child.Type]; ok {
+		child, err := c.constructor(*config.Blacklist.Child, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		b := &Blacklist{
+			paths: config.Blacklist.Paths,
+			s:     child,
+		}
+
+		b.patterns = make([]*regexp.Regexp, len(config.Blacklist.Patterns))
+
+		for i, p := range config.Blacklist.Patterns {
+			re, err := regexp.Compile(p)
+			if err != nil {
+				return nil, fmt.Errorf("Invalid regular expression: %s", p)
+			}
+			b.patterns[i] = re
+		}
+
+		return b, nil
+	}
+
+	return nil, ErrInvalidMetricOutputType
+}
+
+//------------------------------------------------------------------------------
+
+// allowPath checks whether or not a given path is in the allowed set of
+// paths for the Blacklist metrics stat.
+func (h *Blacklist) rejectPath(path string) bool {
+	for _, p := range h.paths {
+		if strings.HasPrefix(path, p) {
+			return true
+		}
+	}
+	for _, pat := range h.patterns {
+		if pat.MatchString(path) {
+			return true
+		}
+	}
+	return false
+}
+
+//------------------------------------------------------------------------------
+
+// GetCounter returns a stat counter object for a path.
+func (h *Blacklist) GetCounter(path string) StatCounter {
+	if h.rejectPath(path) {
+		return DudStat{}
+	}
+	return h.s.GetCounter(path)
+}
+
+// GetCounterVec returns a stat counter object for a path with the labels
+// discarded.
+func (h *Blacklist) GetCounterVec(path string, n []string) StatCounterVec {
+	if h.rejectPath(path) {
+		return fakeCounterVec(func() StatCounter {
+			return DudStat{}
+		})
+	}
+	return h.s.GetCounterVec(path, n)
+}
+
+// GetTimer returns a stat timer object for a path.
+func (h *Blacklist) GetTimer(path string) StatTimer {
+	if h.rejectPath(path) {
+		return DudStat{}
+	}
+	return h.s.GetTimer(path)
+}
+
+// GetTimerVec returns a stat timer object for a path with the labels
+// discarded.
+func (h *Blacklist) GetTimerVec(path string, n []string) StatTimerVec {
+	if h.rejectPath(path) {
+		return fakeTimerVec(func() StatTimer {
+			return DudStat{}
+		})
+	}
+	return h.s.GetTimerVec(path, n)
+}
+
+// GetGauge returns a stat gauge object for a path.
+func (h *Blacklist) GetGauge(path string) StatGauge {
+	if h.rejectPath(path) {
+		return DudStat{}
+	}
+	return h.s.GetGauge(path)
+}
+
+// GetGaugeVec returns a stat timer object for a path with the labels
+// discarded.
+func (h *Blacklist) GetGaugeVec(path string, n []string) StatGaugeVec {
+	if h.rejectPath(path) {
+		return fakeGaugeVec(func() StatGauge {
+			return DudStat{}
+		})
+	}
+	return h.s.GetGaugeVec(path, n)
+}
+
+// SetLogger sets the logger used to print connection errors.
+func (h *Blacklist) SetLogger(log log.Modular) {
+	h.s.SetLogger(log)
+}
+
+// Close stops the Statsd object from aggregating metrics and cleans up
+// resources.
+func (h *Blacklist) Close() error {
+	h.s.Close()
```

Comment:
There might be an error returned from the child so change this to `return h.s.Close()`